packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -122,31 +122,46 @@ function processPositionalParams(renderNode, positionalParams, params, attrs) {
   // if the component is rendered via {{component}} helper, the first
   // element of `params` is the name of the component, so we need to
   // skip that when the positional parameters are constructed
-  const paramsStartIndex = renderNode.getState().isComponentHelper ? 1 : 0;
   const isNamed = typeof positionalParams === 'string';
-  let paramsStream;
 
   if (isNamed) {
-    paramsStream = new Stream(() => {
-      return readArray(params.slice(paramsStartIndex));
-    }, 'params');
-
-    attrs[positionalParams] = paramsStream;
+    processRestPositionalParameters(renderNode, positionalParams, params, attrs);
+  } else {
+    processNamedPositionalParameters(renderNode, positionalParams, params, attrs);
   }
+}
 
-  if (isNamed) {
-    for (let i = paramsStartIndex; i < params.length; i++) {
-      let param = params[i];
-      paramsStream.addDependency(param);
-    }
-  } else {
-    for (let i = 0; i < positionalParams.length; i++) {
-      let param = params[paramsStartIndex + i];
+function processNamedPositionalParameters(renderNode, positionalParams, params, attrs) {
+  const paramsStartIndex = renderNode.getState().isComponentHelper ? 1 : 0;
+
+  for (let i = 0; i < positionalParams.length; i++) {
+    let param = params[paramsStartIndex + i];
+    if (!(positionalParams[i] in attrs)) {
```

Comment:
Can you make this an assertion though?

Basically:

```javascript
assert(`You cannot specify both a positional param (at position ${i}) and the hash argument \`${positionalParams[i]}\`.`, positonalParams[i] in attrs);
```

Then the tests get updated to something like:

```javascript
expectAssertion(function() {
  runAppend(view);
}, /The message from above, I am too lazy to type/);
```